strongbox-commons/src/main/java/org/carlspring/strongbox/config/PropertiesPathResolverConfig.java

```diff
@@ -0,0 +1,106 @@
+         * {@code 'file:${strongbox.home}'} (unless there is a default path prefix override specified in the
+         * {@link #PREFIX_OVERRIDE_PROPERTY} property).
+         *
+         * @param customPathPropertyName the name of the property that may hold a user specified path
+         * @param defaultPath default path to use if no user specified path is set in the custom path property
+         * @return final path to use to lookup the properties resource - including the proper prefix ('file://' or 'classpath:')
+         */
+        public String resolve(final String customPathPropertyName, final String defaultPath)
+        {
+            final String customPathPropertyValue = environment.getProperty(customPathPropertyName);
+            if (!StringUtils.isEmpty(customPathPropertyValue))
+            {
+                final String resolved = customPathPropertyValue.startsWith("classpath:") || customPathPropertyValue.startsWith("file:") ?
+                                        customPathPropertyValue : "file://" + ensureAbsolute(customPathPropertyValue);
+                logger.debug("Resolved to path '{}' using custom property '{}'.", resolved, customPathPropertyName);
+                return resolved;
+            }
+
+            // Use 'file://${strongbox.home}' as prefix unless an override has been set
+            final String pathPrefix;
+            final String defaultPathPrefixOverride = environment.getProperty(PREFIX_OVERRIDE_PROPERTY);
+            if (StringUtils.isEmpty(defaultPathPrefixOverride))
+            {
+                pathPrefix = "file://" + getStrongboxHome();
+            }
+            else
+            {
+                pathPrefix = defaultPathPrefixOverride;
+            }
+            logger.debug("Using pathPrefix '{}'.", pathPrefix);
+
+            final String resolved = pathPrefix + defaultPath;
+            logger.debug("Resolved to path '{}' using default '{}' - no custom path set in '{}'.", resolved, defaultPath, customPathPropertyName);
+            return resolved;
+        }
+
+        private String ensureAbsolute(final String path) {
+
+            if (path.startsWith("..") ) {
+                final String currentDirAbs = Paths.get(".").toAbsolutePath().normalize().toString();
+                logger.debug("Path started with relative parent dir '..'  - converted to absolute path '{}/..'.", currentDirAbs);
+                return currentDirAbs + "/.." + (path.length() > 2 ? path.substring(2) : "");
+            }
+
+            if (path.startsWith(".")) {
+                final String currentDirAbs = Paths.get(".").toAbsolutePath().normalize().toString();
+                logger.debug("Path started with relative current dir '.'  - converted to absolute path '{}'.", currentDirAbs);
+                return currentDirAbs + (path.length() > 1 ? path.substring(1) : "");
+            }
+
+            return path;
```

Comment:
will this method ensure that `etc/conf/conf.yaml` is absolute ? If `etc/conf/conf.yaml` is provided then `etc/conf/conf.yaml` is returned which is not absolute

Reply:
I have renamed it as the intention of the method is to ensure that if it begins w/ `.` or `..` (current or parent dir) that it expands those as `FileUriResource` does not handle those well. 

Need to protect against the cases where a user-specified custom properties path (via the custom property) or a `strongbox.home` has been defined w/ a `.` or `..`